index.js

```diff
@@ -110,22 +110,44 @@ function verify(keys, sig, msg) {
 
 // OTHER CRYTPO FUNCTIONS
 
-exports.signObj = function (keys, hmac_key, obj) {
-  if (!obj) (obj = hmac_key), (hmac_key = null);
-  var _obj = clone(obj);
+function validateParameters(obj, hmac_key) {
+  var temp;
+
+  if (isObject(hmac_key) && (isBuffer(obj) || isString(obj) || obj === null)) {
```

Comment:
One thing I noticed is that the old API supported `verifyObj(keys, undefined, hmac_key)` but if you're checking for a `null`, then the undefined case will not go inside this `if`, and that would be a breaking change. So I think it's safer to just check if the second argument is falsy.

By the way, it would be easier to understand this function's implementation if the arguments were called `first` and `second`.

Reply:
Also, to be sure, it would be good to have tests for all these weird cases. (In fact, good to have tests for these weird cases **before** this PR #80 is merged, to ensure backwards compat)

Reply:
Ha! I had `undefined` in my initial code and I didn't keep it because I couldn't find a usage example with `undefined`.

I'll tidy up the tests and update the PR.

Reply:
Good @davegomez, also note it would be useful to have a separate PR for tests that should be true for the current `main` branch codebase.

Reply:
> By the way, it would be easier to understand this function's implementation if the arguments were called first and second.

While doing this I thought that is important for the function to differentiate what we are doing here. Because this is not a generalistic function used to swap two parameters but is instead of checking for very specific conditions for each of the parameters.

I added a comment and docs to clarify what is happening instead.